opencensus/trace/span.py

```diff
@@ -12,20 +12,113 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+try:
+    from collections.abc import MutableMapping
+    from collections.abc import Sequence
+except ImportError:
+    from collections import MutableMapping
+    from collections import Sequence
+
+from collections import OrderedDict
+from collections import deque
 from datetime import datetime
 from itertools import chain
 
 from opencensus.common import utils
-from opencensus.trace import attributes
+from opencensus.trace import attributes as attributes_module
 from opencensus.trace import base_span
 from opencensus.trace import link as link_module
-from opencensus.trace import stack_trace
-from opencensus.trace import status
-from opencensus.trace import time_event as time_event_module
+from opencensus.trace import stack_trace as stack_trace_module
+from opencensus.trace import status as status_module
+from opencensus.trace import time_event
 from opencensus.trace.span_context import generate_span_id
 from opencensus.trace.tracers import base
 
 
+# https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/TraceConfig.md  # noqa
+MAX_NUM_ATTRIBUTES = 32
+MAX_NUM_ANNOTATIONS = 32
+MAX_NUM_MESSAGE_EVENTS = 128
+MAX_NUM_LINKS = 32
+
+
+class BoundedList(Sequence):
```

Comment:
Nit: consider splitting these helper data structures to another file.

Reply:
I know they're generic, but I left them in here for now since `span` is the only caller.